ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/attestation/ValidateableAttestation.java

```diff
@@ -50,15 +57,38 @@ private ValidateableAttestation(
     this.hashTreeRoot = Suppliers.memoize(attestation::hash_tree_root);
   }
 
+  public Optional<IndexedAttestation> getMaybeIndexedAttestation() {
+    return maybeIndexedAttestation;
+  }
+
   public IndexedAttestation getIndexedAttestation() {
     return maybeIndexedAttestation.orElseThrow(
         () ->
             new UnsupportedOperationException(
                 "ValidateableAttestation does not have an IndexedAttestation yet."));
   }
 
-  public void setIndexedAttestation(IndexedAttestation maybeIndexedAttestation) {
-    this.maybeIndexedAttestation = Optional.of(maybeIndexedAttestation);
+  public Optional<Bytes32> getMaybeRandaoMix() {
+    return maybeRandaoMix;
+  }
+
+  public Bytes32 getRandaoMix() {
+    return maybeRandaoMix.orElseThrow(
+        () ->
+            new UnsupportedOperationException(
+                "ValidateableAttestation does not have a randao mix yet."));
+  }
+
+  public void setIndexedAttestation(IndexedAttestation indexedAttestation) {
+    this.maybeIndexedAttestation = Optional.of(indexedAttestation);
+  }
+
+  public void saveRandaoMix(BeaconState state) {
+    UInt64 randaoIndex =
+        compute_epoch_at_slot(attestation.getData().getSlot())
+            .plus(EPOCHS_PER_HISTORICAL_VECTOR - MIN_SEED_LOOKAHEAD - 1);
+    Bytes32 mix = get_randao_mix(state, randaoIndex);
+    this.maybeRandaoMix = Optional.of(mix);
```

Comment:
Didn't we circle back round to storing the seed? Sorry, I know I said randao mix but I was overthinking the performance implications I think.  As long as we don't have to calculate a seed for every attestation when producing blocks it will be fine.

Reply:
Done.